library/VTKExtensions/Readers/vtkF3DMemoryMesh.cxx

```diff
@@ -0,0 +1,74 @@
+#include "vtkF3DMemoryMesh.h"
+
+#include "vtkDataArrayRange.h"
+#include "vtkFloatArray.h"
+#include "vtkInformationVector.h"
+#include "vtkObjectFactory.h"
+#include "vtkPointData.h"
+#include "vtkSMPTools.h"
+
+#include <numeric>
+
+vtkStandardNewMacro(vtkF3DMemoryMesh);
+
+//------------------------------------------------------------------------------
+vtkF3DMemoryMesh::vtkF3DMemoryMesh()
+{
+  this->SetNumberOfInputPorts(0);
+}
+
+//------------------------------------------------------------------------------
+vtkF3DMemoryMesh::~vtkF3DMemoryMesh() = default;
+
+//------------------------------------------------------------------------------
+void vtkF3DMemoryMesh::SetPoints(const std::vector<float>& positions)
+{
+  vtkNew<vtkFloatArray> arr;
+  arr->SetNumberOfComponents(3);
+  arr->SetNumberOfTuples(positions.size() / 3);
+
+  std::copy(positions.cbegin(), positions.cend(), arr->GetPointer(0));
```

Comment:
I guess thats not SMP accelerated ? Should it be ?

Reply:
Its not necesarry, I'm just wondering if there is a reason for you not to try to do it.

Reply:
I saw no smp copy functions.  
What about getting rid of smp entirely and use C++17 parallel algorithms?

Reply:
I pushed a commit for C++17 algorithms everywhere instead of VTK SMP, let me know what you think.

Reply:
ill check but sad to not use tbb when we can

Reply:
Well, as long as it's parallelized, we don't care if we are using STD or TBB, do we?

Reply:
Well, it looks like parallel algorithms are not well supported by some compilers, including Apple Clang...  
I'll revert these changes.

Reply:
we do, TBB performs better.

Reply:
Well, for copying data, I'm suspect it wouldn't make a difference.  
Anyway, I migrated toward `vtkSMPTools::For`. I can't use `vtkSMPTools::Transform` because it's not available in VTK 9.0